uom_crm_dna.py

```diff
@@ -1,83 +1,235 @@
+
+	# All reference files in a list
+	reference_files = glob(f'reference_standards/TruQ1_RS_*.txt')
+
+	for file in reference_files:
+
+		with open(file) as reference:
+
+			for line in reference:
+
+				x = line.split('\t')
+
+				# Concatenate position and base change
+				v = x[3] + ":" + x[4]	
+
+				# 7 is vaf			
+				reference_variants_dict[v] = float(x[7])  
+
+			##THIS PUTS BOTH REFERENCE FILES IN 1 DICT NOT 2 WHICH IS NEEDED!!
+			return reference_variants_dict 
+
+	#get_variants_in_reference_files(reference_files)    		
+
+
+
+def compare_files(filepaths, reference_variants, split_sample_filepaths):
+	"""
+	Comparing dictionaries with the reference standard dictionary
+ 	"""	
+ 
+		# Key is chr and pos, value is vaf	
+		for key, value in reference_variants_dict.items():
+
+		# Worksheet that corresponds to that run
+		worksheet = subprocess.run([f'grep {sample_id} /data/archive/*/{run_id}/SampleSheet.csv | cut -f 2 -d ","'], shell=True, capture_output=True)
+		
+			if key in samples_dict.keys():
+
+				difference = float(samples_dict[key]) - value
+
+			# Key is variant position, samples_dict[key] is sample vaf, difference is difference between expected and observed vaf, value is RS value, difference/value is % difference
+				print(run_id\
+					, worksheet.stdout.rstrip()\
+					, sample_id, key\
+					, float(samples_dict[key])\
+					, value\
+					, (sigfig.round(difference, sigfigs=3))\
+					, (sigfig.round((difference/value)*100, sigfigs=3))\
+					, "True")
+
+			else:
+
+				print(run_id\
+					, worksheet.stdout.rstrip()\
+					, sample_id, key\
+					, ''\
+					, "n/a"\
+					, "n/a"\
+					, "n/a"\
+					, "n/a"\
+					, "False")
+
+
+###########################
+###	     Programme	    ###
+###########################
+
+if __name__ == '__main__':
+
+#	# Arguments
+	parser = argparse.ArgumentParser()
+	parser.add_argument('--config_file', 'config', help = 'yaml config file')
+	parser.add_argument('--reference_dict', 'r', help = 'dictionary of sample IDs with rs')   
+	parser.add_argument('--filepath_list', 'fl', help = 'list of sample filepaths')
+	parser.add_argument('--file', 'f', help = 'splitting the filepaths into run id and sample id')
+	parser.add_argument('--reference_variants', 'ref', help = 'dict of reference variants/vaf')
+	
+	args = parser.parse_args()
+
+##THESE SHOULD BE SAME NAME AS IN RETURN COMMANDS?? 
+
+# Load uncertainty of measurement sample IDs and their corresponding reference standard into a dictionary
+config_dict = parse_crm_samples(args.[config_file])
+
+# Get all uom sample files into list
+filepath_list = crm_sample_file_list(args.reference_dict)
+
+# Get all variants and their corresponding vaf in a dictionary
+sample_variant_dict = get_variants_in_files_crm(args.filepath_list) #OR# get_variants_in_file(args.tsvs_to_analyse)	
 
+# Split the sample filepaths into sample id and run id
+split_paths = split_sample_filepaths_crm(args.file) #OR# split_sample_filepaths(args.tsvs_to_analyse)
```

Comment:
This isnt doing anything at the mo, should it be called within `get_variants_in_files_crm` instead and saved within the dictionary created within there?